lib/global_id/railtie.rb

```diff
@@ -15,13 +15,13 @@ class Railtie < Rails::Railtie # :nodoc:
 
     initializer 'global_id' do |app|
 
-      app.config.global_id.app ||= app.railtie_name.remove('_application').dasherize
-      GlobalID.app = app.config.global_id.app
+      config.after_initialize do
+        app.config.global_id.app ||= app.railtie_name.remove('_application').dasherize
+        GlobalID.app = app.config.global_id.app
```

Comment:
This will break GlobalIDs generated in initializers themselves, however.

Perhaps we should set default config and assign to GlobalID in the initializer, then assign to GlobalID again after initialize to pick up app config.

Reply:
@jeremy - Ohh yeah that's something that I missed!

> Perhaps we should set default config and assign to GlobalID in the initializer, then assign to GlobalID again after initialize to pick up app config.

Probably the only option we have. I've updated the code to reflect this behavior.